homework/index.js

```diff
@@ -30,17 +15,119 @@
+      text: repository.name,
+      class: 'value',
     });
+
+    // raw 2
+    const tr2 = createAndAppend('tr', repInfoTable);
+    createAndAppend('td', tr2, { class: 'heading', text: 'Description:' });
+    createAndAppend('td', tr2, { text: repository.description, class: 'value' });
+
+    // raw 3
+    const tr3 = createAndAppend('tr', repInfoTable);
+    createAndAppend('td', tr3, { class: 'heading', text: 'Forks:' });
+    createAndAppend('td', tr3, { text: repository.forks, class: 'value' });
+
+    // raw 4
+    const tr4 = createAndAppend('tr', repInfoTable);
+    createAndAppend('td', tr4, { class: 'heading', text: 'Updated:' });
+    createAndAppend('td', tr4, { text: repository.updated_at, class: 'value' });
   }
 
+  // display the contributors of the selected repository ( right side )-----------------/ function /
+  function showContributors(container, contributors) {
+    createAndAppend('h2', container, { text: 'Contributions' });
+
+    const ul = createAndAppend('ul', container);
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      const a = createAndAppend('a', li, {
+        href: contributor.html_url,
+        target: '_blank',
+        text: contributor.login,
+      });
+      createAndAppend('img', a, { src: contributor.avatar_url, width: 100, class: 'spacing' });
+    });
+  }
+
+  // ------------------------------------------------------------------------------------ / function main /
+  function main(url) {
+    const root = document.getElementById('root');
+    // the main header.
+    createAndAppend('h1', root, { text: 'HYF Repositories' });
+
+    // -------------------------------------- ( fetch ALL DATA )
+    fetch(url)
+      .then(Response => Response.json())
+      .then(data => {
+        // the container
+        const theContainer = createAndAppend('div', root, { id: 'container' });
+
+        // the basic repository information (left aside).
+        const basicRepoInfoAside = createAndAppend('aside', theContainer, { class: 'box' });
+        // select element.
+        const select = createAndAppend('select', theContainer);
+
+        // the repository contributors (right aside).
+        const contributorsAside = createAndAppend('aside', theContainer, { class: 'box' });
+
+        // sorting the repositories (case-insensitive) on repository name.
+        data.sort((a, b) => a.name.localeCompare(b.name));
+
+        // appending repositories as options to the select element.
+        data.forEach((repo, index) => {
+          createAndAppend('option', select, { text: repo.name, value: index });
+        });
+
+        // show the first repository information on the (left aside).
+        showRepoBasicInfo(basicRepoInfoAside, data[0]);
+
+        // -------------------------------------------------------- ( fetch contributors for 1st repo )
+        // show the 1st repository contributors on the (right aside).
+        fetch(data[0].contributors_url)
```

Comment:
In addition, if you remember from our class last Sunday, we talked about promise chaining instead of having nested promises. The way you are trying to fetch the contributors details for the first repository, it is being done using nested promises. Can you try to make use of promise chaining here instead of nested promises?

Something like this:
```
// (fetch ALL DATA)
fetch(url)
  .then(resp => resp.json())
  .then(data => {
    // create HTML as you are doing already
   // add the select event listener here
  // possibly return something here for the next .then block to use ;)
  })
  .then(/* fetch contributors details here by calling function created after above comment */)
  .catch(/* this need not be touched as it is being done correctly */);
```